src/theory/strings/type_enumerator.h

```diff
@@ -125,6 +120,21 @@ class StringEnumeratorLength {
   }
 
   bool isFinished() { return d_curr.isNull(); }
+
+ private:
+  /** The type we are enumerating */
+  TypeNode d_type;
+  /** The cardinality of the alphabet */
+  uint32_t d_cardinality;
+  /** The data (index to members) */
+  std::vector<unsigned> d_data;
```

Comment:
I think that this should become a vector of nodes (single char strings for strings) and `mkCurr` should call `mkWord()`.

Reply:
I think this will actually stay a vector of indices, where the difference is that a Node vector of characters will be generated on demand.  E.g. for sequences, there will be a level of indirection.

Again, this is something I could make more progress towards but it will require more substantial changes to this class.